azext_iot/operations/hub.py

```diff
@@ -2070,6 +2070,10 @@ def _iot_device_send_message(
     import ssl
     import os
 
+    device = _iot_device_show(target, device_id)
+    if device is not None and device.get("authentication", {}).get("type", "") != "sas":
```

Comment:
Can be simplified to `if device and device.get("authentication", {}).get("type", "") != "sas":`. Also ideally we can compare against an enumeration value rather than a literal string.